build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -193,7 +194,8 @@ class BuildRunnerDaemonBuilder implements DaemonBuilder {
         builders, daemonOptions.builderConfigOverrides,
         isReleaseBuild: daemonOptions.isReleaseBuild);
 
-    var graphEvents = PackageGraphWatcher(packageGraph,
+    // Only actually used for the AutoChangeProvider.
+    Stream<List<WatchEvent>> graphEvents() => PackageGraphWatcher(packageGraph,
```

Comment:
Why even create it then? Should we move the conditional above?

Reply:
I moved it into a function here so it only gets lazily created if its invoked